source/slang/emit.cpp

```diff
@@ -6098,6 +6138,10 @@ struct EmitVisitor
             emitIRGlobalVar(ctx, (IRGlobalVar*) inst);
             break;
 
+        case kIROp_GlobalParam:
+            emitIRGlobalParam(ctx, (IRGlobalParam*) inst);
```

Comment:
There is an argument that using static_cast here although is more typing is preferable to C type casting, in that it validates that cast is possible.

Reply:
Fair point, although I really dislike the tendency of C++ to take concise code and make it a bloated mess. :)

In this case I was mostly just following the precedent of the existing code in the function (my usually preference to have my diffs blend in rather than rock the boat).